src/codegen/mod.rs

```diff
@@ -436,58 +628,58 @@ impl CodeLocation for Expression {
     fn loc(&self) -> pt::Loc {
```

Comment:
I am not convinced this would be a net win. This is purely a syntax change; it has no affect on the runtime.

Now all the enum literals will have to be much longer because they needed the nested `ExpressionType`. Also `ExpressionType` is not a good name.

Reply:
> I am not convinced this would be a net win. This is purely a syntax change; it has no affect on the runtime.

There is no match statement anymore, so it is less code for us. I don't know if rust implements matches differently than a structGEP to argue about the runtime. According to [this answer](https://stackoverflow.com/questions/36920427/how-is-match-implemented-in-a-language-like-rust), match is a jump table followed by a struct GEP.
 
> Now all the enum literals will have to be much longer because they needed the nested `ExpressionType`. 

There will be no enum anymore. I suggested that we use a struct. The enum would tell use the type of the expression. Common fields will be methods of the struct, so there would be no need to extensive match statements.

> Also `ExpressionType` is not a good name.

That is not difficult to solve.